pkg/apis/application/v1alpha1/applicationset_types.go

```diff
@@ -783,7 +784,15 @@ const (
 // ApplicationSetApplicationStatus contains details about each Application managed by the ApplicationSet
 type ApplicationSetApplicationStatus struct {
 	// Application contains the name of the Application resource
-	Application string `json:"application" protobuf:"bytes,1,opt,name=application"`
+	Application string       `json:"application" protobuf:"bytes,1,opt,name=application"`
+	UID         types.UID    `json:"uid" protobuf:"bytes,2,opt,name=uid,casttype=k8s.io/kubernetes/pkg/types.UID"`
+	CreatedAt   *metav1.Time `json:"createdAt,omitempty" protobuf:"bytes,3,opt,name=createdAt"`
+
+	ProgressiveSync *ProgressiveSyncStatus `json:"progressiveSyncStatus,omitempty" protobuf:"bytes,4,opt,name=progressiveSyncStatus"`
+	Health          HealthStatus           `json:"health,omitempty" protobuf:"bytes,5,opt,name=health"`
+}
+
+type ProgressiveSyncStatus struct {
```

Comment:
Would it be possible to flatten this back out? Nesting the progressive sync fields introduces a world of backwards-compatibility pain.

Reply:
Sure! However, is there another way to clarify what fields are specifically for the progressive sync? I fear the `Status` and `Message` that are specifically for the progressive sync will get confused with the App's `Status` and `Message` which are under `Health`

Reply:
You could prefix the field names with `ProgressiveSync` but leave the `json` tags as-is. That would clarify the Go code without breaking the k8s or Argo API contracts.

Reply:
Done!